docs/kinesis/kinesis_connector.md

```diff
@@ -0,0 +1,103 @@
+# Exasol Kinesis Connector 
+
+Exasol Kinesis Connector provides UDF scripts which allow users import data
+from a Kinesis Stream to an Exasol table.
+
+## Prerequisites
+
+* A running Exasol cluster with a version 6.0 or later.
+* An AWS account with all necessary permissions to read from a Kinesis Stream.
+* An AWS Kinesis producer which sends the data to a stream in the valid JSON
+ format. 
+ An example of a valid JSON string: 
+ `{"sensorId": 17,"currentTemperature": 147,"status": "WARN"}`.
+ An example of an invalid JSON format:
+ `{"sensorId": 17,"currentTemperature": 147,"status": "WARN",}`. The Exasol
+ Kinesis Connector will not parse this string correctly.
+
+## Deployment
+
+Please refer to the [Deployment Guide](../deployment_guide.md) till the section
+`Upload the JAR file to the bucket` (inclusive) to prepare a bucket and a JAR.
+
+## Preparing a table for data
+
+To store the data from a stream we need a table. 
+The columns in the Exasol table have to imitate the data types of the data
+ stored in the stream and also be in the exact same order.
+ 
+The table also requires two additional columns to store the Kinesis metadata:  
+    `kinesis_shard_id VARCHAR(2000), 
+     kinesis_sequence_number VARCHAR(2000)`
+These two columns must be in the end of the table.
+ 
+For example, we have a stream with the following data: 
+ `{"sensorId": 17,"currentTemperature": 147,"status": "WARN"}`
+We create a table for this stream:
+
+```sql
+OPEN SCHEMA <schema_name>;
+
+CREATE OR REPLACE TABLE <schema_name>.<table_name> 
+    (sensorId DECIMAL(18,0), 
+    currentTemperature DECIMAL(18,0), 
+    status VARCHAR(100), 
+-- Required for importing data from Kinesis
+    kinesis_shard_id VARCHAR(2000), 
+    kinesis_sequence_number VARCHAR(2000));
+
+```
+
+## Create ETL UDFs Scripts
+
+Create the following UDF scripts. Please do not change the names of the scripts.
+
+```sql
+--/
+CREATE OR REPLACE JAVA SET SCRIPT READ_KINESIS_SHARDS_METADATA (...) EMITS (kinesis_shard_id VARCHAR(2000)) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-0.6.2-SNAPSHOT.jar;
+/
+;
+  
+--/
+CREATE OR REPLACE JAVA SET SCRIPT IMPORT_KINESIS_SINGLE_SHARD_DATA (...) EMITS (...) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisShardDataImporter;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-0.6.2-SNAPSHOT.jar;
+/
+;
+    
+--/
+CREATE OR REPLACE JAVA SET SCRIPT GENERATE_KINESIS_IMPORT_QUERY (...) EMITS (...) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisImportQueryGenerator;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-0.6.2-SNAPSHOT.jar;
+/
+; 
```

Comment:
Rename the UDF name into: `KINESIS_METADATA`, `KINESIS_IMPORT` and `KINESIS_PATH`.

Reply:
It's very misleading in my eyes that I call `KINESIS_PATH` script to run an import command. I used these names first, but they bothered me so much that I renamed them to make it easier to understand what each of them does